impl/client/src/main/javascript/web/dojo/pentaho/common/overrides/dojo/store/Memory.js

```diff
@@ -0,0 +1,224 @@
+define(["../_base/declare", "./util/QueryResults", "./util/SimpleQueryEngine" /*=====, "./api/Store" =====*/],
+function(declare, QueryResults, SimpleQueryEngine /*=====, Store =====*/){
+
+// module:
+//		dojo/store/Memory
+
+// No base class, but for purposes of documentation, the base class is dojo/store/api/Store
+var base = null;
+/*===== base = Store; =====*/
+
+return declare("dojo.store.Memory", base, {
+	// summary:
+	//		This is a basic in-memory object store. It implements dojo/store/api/Store.
+	constructor: function(options){
+		// summary:
+		//		Creates a memory object store.
+		// options: dojo/store/Memory
+		//		This provides any configuration information that will be mixed into the store.
+		//		This should generally include the data property to provide the starting set of data.
+		for(var i in options){
+			this[i] = options[i];
+		}
+		this.setData(this.data || []);
+	},
+	// data: Array
+	//		The array of all the objects in the memory store
+	data:null,
+
+	// idProperty: String
+	//		Indicates the property to use as the identity property. The values of this
+	//		property should be unique.
+	idProperty: "id",
+
+	// index: Object
+	//		An index of data indices into the data array by id
+	index:null,
+
+	// queryEngine: Function
+	//		Defines the query engine to use for querying the data store
+	queryEngine: SimpleQueryEngine,
+	get: function(id){
+		// summary:
+		//		Retrieves an object by its identity
+		// id: Number
+		//		The identity to use to lookup the object
+		// returns: Object
+		//		The object in the store that matches the given id.
+		return this.data[this.index[id]];
+	},
+	getIdentity: function(object){
+		// summary:
+		//		Returns an object's identity
+		// object: Object
+		//		The object to get the identity from
+		// returns: Number
+		return object[this.idProperty];
+	},
+	put: function(object, options){
+		// summary:
+		//		Stores an object
+		// object: Object
+		//		The object to store.
+		// options: dojo/store/api/Store.PutDirectives?
+		//		Additional metadata for storing the data.  Includes an "id"
+		//		property if a specific id is to be used.
+		// returns: Number
+		var data = this.data;
+		var index = this.index;
+		var idProperty = this.idProperty;
+		var id = object[idProperty] = (options && "id" in options) ?
+			options.id : idProperty in object ? object[idProperty] : Math.random();
```

Comment:
[comment]: <> (FrogbotReviewComment)


## 🎯 Static Application Security Testing (SAST) Vulnerability
<div align='center'>

| Severity                | Finding                  |
| :---------------------: | :-----------------------------------: |
| ![](https://raw.githubusercontent.com/jfrog/frogbot/master/resources/v2/applicableLowSeverity.png)<br>     Low | Use of Insecure Random |

</div>

<details>
<summary> <b>Full description</b> </summary>
<br>


### Overview
A use of insecure random vulnerability is a type of security flaw that is
caused by the use of inadequate or predictable random numbers in a program
or system. Random numbers are used in many security-related applications,
such as generating cryptographic keys and if the numbers are not truly
random, an attacker may be able to predict or recreate them, potentially
compromising the security of the system.

### Vulnerable example
```javascript
var randomNum = Math.random();
```
`Math.random` is not secured, as it creates predictable random numbers.

### Remediation
```diff
var randomNum = crypto.randomInt(0, 100)
```
`crypto.randomInt` is secured, and creates much less predictable random
numbers.


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>